core/src/main/scala/filodb.core/binaryrecord2/RecordSchema.scala

```diff
@@ -419,22 +419,35 @@ object RecordSchema {
                                                        MapColumn -> 4,
                                                        HistogramColumn -> 4)
 
+  private val tlKeyBuf = new ThreadLocal[Array[Byte]]()
```

Comment:
This is a rather odd way if initializing the thread local. There's already a method to do this -- no need for magic unsafe pointer.

Reply:
Did you mean ThreadLocal.withInitial(....)?   That's not easy to use from Scala